### DISCUSSION TITLE
Streams to JointTraining

### DISCUSSION THREAD

**Discussion by evertonaleixo** [ContinualAI/avalanche#991] [Feature Request]
As discussed in #987 the streams generated in benchmarks are not appropriated to JointTrainingStrategy, because it only concatenates the data set of the training stream, and it is done in an internal function. However, eval and validation streams still are considered as many experiences. As an upper bound, it should take its metrics in the union of the data sets too. Besides that, if the validation stream has many experiences, the early stop plugin does not know which experience to use to stop.

Therefore, I propose to create a util function, to convert the benchmarks into a new benchmark. In this new benchmark, each stream will have just one experience containing all data of all experiences of the stream.

I have already implemented this function to use in my experiments.

Best regards,
Everton Aleixo.

**Comment by AntonioCarta** [comment]
I don't think we already have this feature so feel free to open a PR.

**Comment by evertonaleixo** [comment]
Thank you!
